src/psyclone/psyGen.py

```diff
@@ -4465,6 +4663,227 @@ def _number_handler(self, node, parent):
         return Literal(node.items[0], parent=parent)
 
 
+class Symbol(object):
+    '''
+    Symbol item for the Symbol Table. It contains information about: the name,
+    the datatype, the shape (in row-major order) and the symbol
+    access characterisctics. The symbol access can be:
+        - local: Variable that just exist in the kernel scope.
+        - external: Global variable.
+        - read_arg: Kernel argument which is only read.
+        - write_arg: Kernel argument which is only written.
+        - readwrite_arg: Kernel argument which is read and written.
+
+    :param str name: Name of the symbol.
+    :param str datatype: Data type of the symbol.
+    :param list shape: Shape of the symbol in row-major order (leftmost \
+                       index is contiguous in memory). Each entry represents \
+                       an array dimension. If not None then it holds the \
+                       extent of that dimension. If it is an empy list it \
+                       represents an scalar.
```

Comment:
"...empty list then the symbol represents a scalar." Please also explain that one or more entries of None means that it is an assumed-shape array. I'm not sure what this means in a language-independent context - maybe it's just a pointer?

Reply:
Modified sentence and clarified that the extent is unknown (from PSyIR point of view). 'assumed-shape' is Fortran-specific and 'pointer' is probably too implementation-specific.

Reply:
Great.